source/mainnet/smart-contracts/references/crypto-primitives.rst

```diff
@@ -0,0 +1,22 @@
+.. _crypto-primitives:
+
+========================
+Cryptographic primitives
+========================
+
+V1 smart contracts natively support a number of common cryptographic primitives.
+Compared to implementing the same functionality in the smart contract, using primitives is subtantially cheaper.
+At the same time, smart contract developers benefit from using a single high-quality implementation of the primitives provided by the chain.
+
+Since all contract inputs, as well as the contract state, on the concordium blockchain are public it only reasonable to have primitives that take public inputs.
```

Comment:
```suggestion
Since all contract inputs, as well as the contract state, on the Concordium blockchain are public, it only reasonable to have primitives that take public inputs.
```